### ISSUE TITLE
free reroll vs bennies

### ISSUE THREAD

**Issue by grendel111111** [javierriveracastro/betteroll-swade#531]
Is there a way in BR2 to identify if a reroll comes from a bennie or a free reroll? At the moment when you have Elan and you reroll either through a bennie or a free reroll you get the +2 bonus. Is there a way to apply it only to bennie rerolls?

**Comment by javierriveracastro** [comment]
No, not right now. I will need to be added from the ground, probably with a new action just for benny based rerolls.

**Comment by brunocalado** [comment]
Yes, this would be nice.

**Comment by ddbrown30** [comment]
Added in the latest version. Actions with the `rerollSkillMod` and/or `rerollDamageMod` property can now optionally use the `rerollMode` property. This can be set to either "benny" or "free" and the action will only trigger for the relevant roll. If the value is not present, it will trigger for all rerolls. Elan and No Mercy have both been updated to use this and so now only trigger on benny rerolls.

I just noticed a bug in this, though, which I'll have to look into. TLDR, it only works for the first reroll so theoretically if you have something with both a free reroll bonus and a benny reroll bonus, you would only get one or the other.

**Comment by ddbrown30** [comment]
I've got it sorted. Should be in the next version. That change also fixes the behaviour to line up with the actual rules for Elan/No Mercy which is that the bonus only applies to the roll where a benny was spent i.e. if you spend a benny to reroll and then free reroll, you don't get the +2 on the free reroll.
